notebooks/nyc_taxi_years_correlation.ipynb

```diff
@@ -58,9 +58,9 @@
     }
```

Comment:
<p>Why is there a big slowdown?  Running on a different GPU?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cuspatial/pull/616/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
@taureandyernv , are you aware how did the notebook run? On what machine, with/without memory pool etc.?

Reply:
you're saving the parquet file as a csv :)

Reply:
@taureandyernv do you have an answer to the question about what hardware it was run on, memory pool etc? We need to make sure this is not a real regression.

Reply:
In the future we should always store this information in the notebook.

Reply:
We store it in the notebook testing script we built.  So I'm using 2x GV100s for these runs, 3GB to play with on single GPU.  It will take 3 minutes 30 seconds on 2x RTX8000 48GB each (but single GPU)

Reply:
<p>acutally...this is about right.  This should take 30ish seconds on a tesla based hardware, 3.5 minutes on quadro.  @isvoid, what slowdown are you talking about?</p>

Reply:
If you look at the diff in this PR, you will see the slowdown I am talking about.

The notebook as checked in stores the time taken by the previous run. We should document in the notebook what hardware, RAPIDS and CUDA version, etc. it was run on. Otherwise when people run and they see a slower time they are likely to be confused or surprised.

None of this "this is about right" guesswork. Let's be precise.

Reply:
Here is the slowdown:

old:

```
      "CPU times: user 11.2 s, sys: 6.27 s, total: 17.5 s\n",
      "Wall time: 17.7 s\n"
```

new

```
      "CPU times: user 22.6 s, sys: 14.6 s, total: 37.2 s\n",
      "Wall time: 37.2 s\n"
```

Reply:
That is interesting and I've honestly don't recall seeing it run that fast.  However, we can't yet compare these numbers and say there's a regression, but we can strongly suspect it. When we built this notebook for the 0.14 release, [I did my work on RTX8000s (here is the commit with the ~3.5 minute run)](https://github.com/rapidsai/cuspatial/blob/76f0a2c595dc0a0d245a821923b6a14bb8cbcaaf/notebooks/nyc_taxi_years_correlation.ipynb), ran some quick tests on the GV100s to see how my time could be so different, and only reran it on GV100s with this update.  @thomcom used a different machine that was consistently faster with the PIP algorithm.  It was cuSpatial's PIP that helped me realize the depth of the "by algorithm" performance discrepancies that classes of cards in the same arch family could have.  Looking back through the branches, the notebook hasn't actually been reran since its merge and the CI jenkins jobs for each release may have already been deleted.  This may end up being a @thomcom question, as those 17 second numbers were run on his machine, as per https://github.com/rapidsai/cuspatial/commit/6c3d853334a3f038df8f53c3ae74e3f679e22abb and this commit ([which I could only render](https://github.com/rapidsai/cuspatial/blob/bdc8f16fa4d7b0257867c25b5133ee80d0cd6442/notebooks/nyc_taxi_years_correlation.ipynb) ) .  We may need him to rerun the notebook on that system to get any realistic understanding about if there is a regression and where it could have happened.  @thomcom , may you provide your system details for @harrism and, if you still have that system, rerun the notebook on that system on 22.08 nightlies and provide us numbers?

## A bit about the past: 
While I can talk about what I recall observing then (and could be mistaken), the challenge you're facing comes from the stoppage of testing and record keeping for regressions on notebooks right around 0.14 and notebooks testing didn't start back up until after 0.18, which wasn't nearly as thorough until very recently.  Some libraries reestablished forms of testing and record keeping a bit more whole heartedly than others, [but cuGraph remains a shining example of what might help with system spec tracking with their table at the top](https://github.com/rapidsai/cugraph/blob/branch-22.08/notebooks/algorithms/centrality/Centrality.ipynb)).  For now, would it help that current and future cuSpatial notebooks include a version of cuGraph's tables? 

## Thoughts on the future:
I absolutely agree on the precise record keeping, with system specs, which is why it's a built in portion of the outputs with that new beta notebooks testing script (although we currently don't test the cuSpatial notebook).  We're only collecting data, not yet processing or analyzing it.  We're in talks to get it integrated into Ops.  Would you like me to start testing cuSpatial notebooks?  Happy to share more info.

Reply:
Hey @taureandyernv @harrism I'll try to rerun the notebook this afternoon or tomorrow.